src/ray/raylet/worker_pool.cc

```diff
@@ -455,6 +471,8 @@ void WorkerPool::MonitorStartingWorkerProcess(const Process &proc,
                                           status, &found, &used, &task_id);
       }
       state.starting_worker_processes.erase(it);
+      runtime_env_manager_.RemoveURIReference(kWorkerSetupTokenPrefix +
```

Comment:
Is it guaranteed that this only happens after AddURIReference in OnWorkerStarted?  I just want to make sure there isn't a gap between when the reference is removed here and when it's added in OnWorkerStarted.  If there's a gap, then the refcount might become 0 in the gap and we might start deleting the resources

Reply:
We can guarantee that this only happens after  AddURIReference  in `StartWorkerProcess`, not `OnWorkerStarted `. If some worker thread register timeout after process started, we will run this RemoveURIReference.

Reply:
I think I understand, just to make sure could you just clarify the order of these four events?  (1) AddURI in StartWorkerProcess, (2) RemoveURI in MonitorStartingWorkerProcess, (3) AddURI in OnWorkerStarted, (4) RemoveURI in OnWorkerStarted